### ISSUE TITLE
Rebot generates invalid output.xml when using `--removekeywords` and there's IF on test case level

### ISSUE THREAD

**Issue by mkorpela** [robotframework/robotframework#4009]
Here is the test case:
```RobotFramework
*** Test Cases ***
Testing 2
   IF  1 == 1
       Log  hello
   ELSE
       Log  none
   END
   Keyword

*** Keywords ***
Keyword
   IF  1 == 1
       Fail
   ELSE
       Log  none
   END
```
Then do
`rebot --log NONE --report NONE --xunit NONE --consolecolors off --NoStatusRC --removekeywords all --output output_invalid.xml output.xml`
If you do:
`rebot --log NONE --report NONE --xunit NONE --consolecolors off --NoStatusRC --removekeywords all --output output_invalid.xml output_invalid.xml`
Boom => 

```
[ ERROR ] Reading XML source 'output_invalid.xml' failed: Incompatible child element 'doc' for 'if'.

Try --help for usage information.
```

**Comment by mkorpela** [comment]
Credit for this goes to @emakaay https://github.com/mkorpela/pabot/issues/364

**Comment by pekkaklarck** [comment]
I can reproduce the problem and also have a pretty good idea why it occurs. Luckily the problem only occurs if there's IF on the test case level, not if there's IF in user keyword used by the test. Although this is an annoying bug, I don't think this is severe enough to warrant one more RF 4.0.x release, especially now when RF 4.1 work has just started. This obviously will be fixed RF 4.1.

**Comment by miktuy** [comment]
Hi! Try to fix in https://github.com/robotframework/robotframework/pull/4010

**Comment by pekkaklarck** [comment]
I merged #4010 and it fixes the crash. A problem now is that the IF structure is completely removed from the log file. Considering that all keyword data is removed, we could consider this fine but I'd like to have something left there. Anyway, this is a lot smaller problem than the crash.

**Comment by pekkaklarck** [comment]
The aforementioned problem was fixed in the above commit.
